app/src/main/java/com/github/sdp/tarjetakuna/ui/singlecard/SingleCardFragment.kt

```diff
@@ -0,0 +1,63 @@
+package com.github.sdp.tarjetakuna.ui.singlecard
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import com.github.sdp.tarjetakuna.R
+import com.github.sdp.tarjetakuna.databinding.FragmentSingleCardBinding
+import com.github.sdp.tarjetakuna.model.MagicCard
+import com.github.sdp.tarjetakuna.model.MagicType
+import com.google.gson.Gson
+import com.github.sdp.tarjetakuna.utils.CustomGlide
+
+/**
+ * This fragment is used to display a single card with some details.
+ * To run this fragment, you need to pass a MagicCard as a json string in the arguments with the key "card".
+ * @see MagicCard
+ */
+class SingleCardFragment: Fragment() {
+
+    private var _binding: FragmentSingleCardBinding? = null
+    private val binding get() = _binding!!
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentSingleCardBinding.inflate(inflater, container, false)
+
+        try {
+            val card = Gson().fromJson(arguments?.getString("card"), MagicCard::class.java)
+            binding.singleCardTextCardName.text = card.name
```

Comment:
might be worth it to split this in a separate function to be able to call it not only on onCreateView, but also on Update maybe ?

Reply:
Why do we need it in a potential Update method?

Reply:
its the principle of separation, easier to change later on. If you have the card information that change, you ll have no way of updating the view except recreating it